src/Bundle.ts

```diff
@@ -24,21 +24,56 @@ import { OutputOptions } from './rollup/index';
 import { RawSourceMap } from 'source-map';
 import Graph from './Graph';
 import ExternalModule from './ExternalModule';
+import ExternalVariable from './ast/variables/ExternalVariable';
+import ExportDefaultVariable from './ast/variables/ExportDefaultVariable';
+import { DynamicImportMechanism } from './ast/nodes/Import';
 
 export default class Bundle {
 	graph: Graph;
-	modules: Module[];
 	orderedModules: Module[];
 	externalModules: ExternalModule[];
 	entryModule: Module;
 
-	constructor (graph: Graph, orderedModules: Module[], externalModules: ExternalModule[], entryModule: Module) {
+	constructor (graph: Graph, orderedModules: Module[]) {
 		this.graph = graph;
 		this.orderedModules = orderedModules;
-		this.externalModules = externalModules;
+		this.externalModules = undefined;
+		this.entryModule = undefined;
+	}
+
+	bind () {
+		this.orderedModules.forEach(module => module.bindImportSpecifiers());
+		this.orderedModules.forEach(module => module.bindReferences());
+	}
+
+	includeMarked (treeshake: boolean) {
+		if (treeshake) {
+			let addedNewNodes;
+			do {
+				addedNewNodes = false;
+				this.orderedModules.forEach(module => {
+					if (module.includeInBundle()) {
+						addedNewNodes = true;
+					}
+				});
+			} while (addedNewNodes);
+		} else {
+			// Necessary to properly replace namespace imports
+			this.orderedModules.forEach(module => module.includeAllInBundle());
+		}
+	}
+
+	setFascade (entryModule: Module) {
```

Comment:
One last thing that tripped me that it is probably not clear immediately, what is exactly fascading what here. If I interpret this correctly, you mean that the created bundle file behaves as if it was the entry module file. So maybe this is `setOutputFascade`?